code/modules/deathmatch/deathmatch_modifier.dm

```diff
@@ -0,0 +1,478 @@
+	ADD_TRAIT(lobby, TRAIT_DEATHMATCH_EXPLOSIVE_IMPLANTS, DEATHMATCH_TRAIT)
+
+/datum/deathmatch_modifier/explode_on_death/apply(mob/living/carbon/player, datum/deathmatch_lobby/lobby)
+	var/obj/item/implant/explosive/deathmatch/implant = new()
+	implant.implant(player, silent = TRUE, force = TRUE)
+
+/datum/deathmatch_modifier/helgrasp
+	name = "Helgrasped"
+	description = "Cursed hands are being thrown at you!"
+
+/datum/deathmatch_modifier/helgrasp/apply(mob/living/carbon/player, datum/deathmatch_lobby/lobby)
+	var/metabolism_rate = /datum/reagent/inverse/helgrasp/heretic::metabolization_rate
+	player.reagents.add_reagent(/datum/reagent/inverse/helgrasp/heretic, initial(lobby.map.automatic_gameend_time) / metabolism_rate)
+
+/datum/deathmatch_modifier/wasted
+	name = "Wasted"
+	description = "You've had one drink too many"
+
+/datum/deathmatch_modifier/wasted/apply(mob/living/carbon/player, datum/deathmatch_lobby/lobby)
+	player.adjust_drunk_effect(rand(30, 35))
+	var/metabolism_rate = /datum/reagent/consumable/ethanol/kahlua::metabolization_rate
+	player.reagents.add_reagent(/datum/reagent/consumable/ethanol/kahlua, initial(lobby.map.automatic_gameend_time) * 0.35 / metabolism_rate)
+
+/datum/deathmatch_modifier/monkeys
+	name = "Monkeyfication"
+	description = "Go back, I want to be monkey!"
+
+/datum/deathmatch_modifier/monkeys/apply(mob/living/carbon/player, datum/deathmatch_lobby/lobby)
+	//we don't call monkeyize(), because it'd set the player name to a generic "monkey(number)".
+	player.set_species(/datum/species/monkey)
+
+/datum/deathmatch_modifier/inverted_movement
+	name = "Inverted Movement"
+	description = "Up is down, left is right"
+
+/datum/deathmatch_modifier/inverted_movement/apply(mob/living/carbon/player, datum/deathmatch_lobby/lobby)
+	player.AddElement(/datum/element/inverted_movement)
+
+/datum/deathmatch_modifier/minefield
+	name = "Minefield"
+	description = "Oh, it seems you've trotted on a mine!"
+
+/datum/deathmatch_modifier/minefield/on_start_game(datum/deathmatch_lobby/lobby)
+	var/list/mines = subtypesof(/obj/effect/mine)
+	mines -= list(
+		/obj/effect/mine/explosive, //too lethal.
+		/obj/effect/mine/kickmine, //will kick the client, lol
+		/obj/effect/mine/gas, //Just spawns oxygen.
+	)
+
+	///1 every 10 turfs, but it will actually spawn fewer mines since groundless and closed turfs are skipped.
+	var/mines_to_spawn = length(lobby.location.reserved_turfs) * 0.1
+	for(var/iteration in 1 to mines_to_spawn)
+		var/turf/target_turf = pick(lobby.location.reserved_turfs)
+		if(!isopenturf(target_turf) || isgroundlessturf(target_turf))
+			continue
+		///don't spawn mine next to player spawns.
+		if(locate(/obj/effect/landmark/deathmatch_player_spawn) in range(1, target_turf))
+			continue
+		var/mine_path = pick(mines)
+		new mine_path (target_turf)
+
+/datum/deathmatch_modifier/flipping
+	name = "Perma-Flipping"
+	description = "You're constantly flipping - the mob that's - however it's purely cosmetic"
```

Comment:
```suggestion
	description = "You're constantly flipping - the mob that's - however it's purely cosmetic"
```
I don't think I understand what the hypenated part of this sentence is supposed to say

Reply:
Yeah, me neither. I must have been pretty distracted to write rubbish.